js-html-games/bogscript.js

```diff
@@ -1,7 +1,378 @@
+        alphaLetter.addEventListener('click', function () {
+            typedInput.push(keyboardLetters[i]);
+            type();
+            console.log(typedInput);
+            console.log(keyboardLetters[i]);
+        })
+        if (i < 10) {
+            document.getElementById("one").appendChild(alphaLetter);
+        }
+        else if (i > 18) {
+            document.getElementById("three").appendChild(alphaLetter);
+        }
+        else {
+            document.getElementById("two").appendChild(alphaLetter);
+        }
+    }
+    document.getElementById("three").appendChild(enter);
+    document.getElementById("three").appendChild(back);
+}
+
+function keyDownEvents(rightWord, wordList) {
+    const buttons = document.querySelectorAll('button');
+    document.addEventListener('keydown', function (event) {
+        if (event.code === "Enter") {
+            buttons[buttons.length - 2].style.backgroundColor = "rgb(104, 157, 157)";
+            matchWord(rightWord, typedInput.join(""), wordList)
+            typedInput = [];
+            console.log(typedInput.join(""));
+        }
+        if (event.code === "Backspace") {
+            buttons[buttons.length - 1].style.backgroundColor = "rgb(104, 157, 157)";
+            erase();
+            typedInput.pop();
+            console.log(typedInput);
+        }
+        for (let i = 0; i < keyboardLetters.length; i++) {
+            if (event.code === `Key${alphabet[i]}`) {
+                buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor = "rgb(104, 157, 157)";
+                typedInput.push(alphabet[i]);
+                type();
+                console.log(typedInput);
+                console.log(alphabet[i]);
+            }
+        }
+    })
+    document.addEventListener('keyup', function (event) {
+        if (event.code === "Enter") {
+            buttons[buttons.length - 2].style.backgroundColor = "lightblue";
+        }
+        if (event.code === "Backspace") {
+            buttons[buttons.length - 1].style.backgroundColor = "lightblue";
+        }
+        for (let i = 0; i < keyboardLetters.length; i++) {
+            if (event.code === `Key${alphabet[i]}`) {
+                if (buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor !== "grey") {
+                    buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor = "lightblue";
+                }
+            }
+        }
+    })
+}
+
+function type() {
+    const guessCols = document.querySelectorAll(".letter");
+    if (typedInput.length < 6) {
+        console.log(currentRow);
+        guessCols[(typedInput.length - 1) + (numC * currentRow)].innerText = typedInput[typedInput.length - 1];
+    }
+    else {
+        typedInput = typedInput.slice(0, numC);
+    }
+}
+
+function erase() {
+    const guessCols = document.querySelectorAll(".letter");
+    console.log(typedInput);
+    guessCols[(typedInput.length-1) + (numC * currentRow)].innerText = "";
```

Comment:
if you're doing not simple calculations (like an add or subtract), pull the calculation out into its own variable and use a descriptive name so it can mean something to the reader. this place and on line 125

```
const eraseIndex = typedInput.length-1 + NUM_OF_COLS * currentRow;
guessCols[eraseIndex].innerText = "";
```